include/seqan3/alphabet/composition/cartesian_composition.hpp

```diff
@@ -321,240 +444,185 @@ class cartesian_composition : public pod_tuple<first_component_type, component_t
     //!\}
 
     /*!\name Read functions
+     * \brief All read operations are constant complexity.
      * \{
      */
-    /*!\brief Return the letter combination's numeric value (or "rank") in the alphabet composition.
-     * \par Complexity
-     * Linear in the number of alphabets.
+    /*!\brief Tuple-like access to the contained components.
+     * \tparam index Return the i-th element.
+     * \returns A proxy to the contained element that models the same alphabets concepts and supports assignment.
      */
-    constexpr rank_type to_rank() const
+    template <size_t index>
+    friend constexpr auto get(cartesian_composition & l)
     {
-        return to_rank_impl(positions);
+        using t = meta::at_c<component_list, index>;
```

Comment:
using an index out of range will trigger an infintiy compiler loop here. I think I encountered similar problems. But maybe it was another meta:: functionality. Just check this maybe.

Reply:
So you should add a static_assert